be/src/exec/pipeline/jdbc_scan_operator.cpp

```diff
@@ -0,0 +1,223 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "exec/pipeline/jdbc_scan_operator.h"
+
+#include <sstream>
+
+#include "common/config.h"
+#include "exec/vectorized/jdbc_scanner.h"
+#include "runtime/descriptors.h"
+#include "runtime/jdbc_driver_manager.h"
+#include "util/defer_op.h"
+
+namespace starrocks {
+namespace pipeline {
+
+JDBCScanOperator::JDBCScanOperator(OperatorFactory* factory, int32_t id, int32_t plan_node_id,
+                                   const TJDBCScanNode& jdbc_scan_node, const std::vector<ExprContext*>& conjunct_ctxs,
+                                   int64_t limit)
+        : SourceOperator(factory, id, "jdbc_scan", plan_node_id),
+          _jdbc_scan_node(jdbc_scan_node),
+          _conjunct_ctxs(conjunct_ctxs),
+          _limit(limit),
+          _result_chunks(config::doris_scanner_queue_size) {}
+
+Status JDBCScanOperator::prepare(RuntimeState* state) {
+    SourceOperator::prepare(state);
+    _state = state;
+    _result_tuple_desc = state->desc_tbl().get_tuple_descriptor(_jdbc_scan_node.tuple_id);
+    _start_scanner_thread(state);
+    return Status::OK();
+}
+
+void JDBCScanOperator::close(RuntimeState* state) {
+    if (_scanner_thread) {
```

Comment:
we need set a flag to nodify _scanner_thread cancel scan data

Reply:
> we need set a flag to nodify _scanner_thread cancel scan data

fixed